source/CMakeLists.txt

```diff
@@ -201,10 +190,58 @@ foreach(build ${DEAL_II_BUILD_TYPES})
       ALIAS ${DEAL_II_TARGET_NAME}_${build_lowercase}
       )
   endif()
+endforeach()
+
+#
+# Now, define a "dealii", i.e., "${DEAL_II_TARGET_NAME}", target that can
+# be used by user projects to automatically switch between the release and
+# debug versions of the library depending on the build type.
+#
+# For convenience, we also add our compiler and linker flags to the target.
+# This is useful for user projects that simply want to use our compiler and
+# linker flags and not set up their own.
+#
+# If a downstream project does not want us to override their compiler and
+# linker flags, then they will need to use the "dealii_debug" and
+# "dealii_release" targets directly.
+#
+
+add_library(${DEAL_II_TARGET_NAME} INTERFACE)
+add_dependencies(library ${DEAL_II_TARGET_NAME})
+
+target_compile_flags(${DEAL_II_TARGET_NAME} INTERFACE
+  "$<COMPILE_LANGUAGE:CXX>" "${DEAL_II_CXX_FLAGS}"
+  )
+target_link_flags(${DEAL_II_TARGET_NAME} INTERFACE "${DEAL_II_LINKER_FLAGS}")
+
+foreach(build ${DEAL_II_BUILD_TYPES})
+  string(TOLOWER ${build} build_lowercase)
```

Comment:
I don't think you actually use this variable here.

Reply:
We do in line 296.